scene/3d/xr_nodes.cpp

```diff
@@ -638,20 +638,19 @@ void XROrigin3D::set_current(bool p_enabled) {
 	set_notify_local_transform(current);
 	set_notify_transform(current);
 
-	if (current) {
-		for (int i = 0; i < origin_nodes.size(); i++) {
-			if (origin_nodes[i] != this) {
-				origin_nodes[i]->set_current(false);
-			}
-		}
-	} else {
-		bool found = false;
-		// We no longer have a current origin so find the first one we can make current
-		for (int i = 0; !found && i < origin_nodes.size(); i++) {
-			if (origin_nodes[i] != this) {
-				origin_nodes[i]->set_current(true);
-				found = true;
+	bool should_enabled(!p_enabled);
+	for (int i = 0; i < origin_nodes.size(); ++i) {
+		if (origin_nodes[i] != this) {
+			origin_nodes[i]->current = should_enabled;
+			if (origin_nodes[i]->is_inside_tree()) {
+				// Notify us of any transform changes
```

Comment:
```suggestion
				// Notify us of any transform changes.
```
Per the [comment style](https://docs.godotengine.org/en/latest/contributing/development/code_style_guidelines.html#comment-style-guide).